.pfnci/linux/tests/cuda-rapids.Dockerfile

```diff
@@ -1,9 +1,11 @@
-ARG BASE_IMAGE="rapidsai/rapidsai-core-dev:23.06-cuda11.8-devel-ubuntu22.04-py3.10"
+ARG BASE_IMAGE="rapidsai/base:23.12-cuda12.0-py3.10"
 FROM ${BASE_IMAGE}
 
+USER root
```

Comment:
Use root user to write to the cache directory that will be mounted to be owned by the host user.
```
ccache: error: Failed to create temporary file for /tmp/cupy_cache/.ccache/tmp/tmp.cpp_stdout.gnumMQ: Permission denied
```